### PR TITLE
api: add variables to endpoint url of openapi spec, fixes #19

### LINKED ISSUE #19: OpenAPI Spec update for server variable
The server parameter has a variable which should be expanded per OAS3.

Here's an example from the Swagger.io documentation:

https://swagger.io/specification/#server-object-example

```yaml
servers:
- url: https://{username}.gigantic-server.com:{port}/{basePath}
  description: The production API server
  variables:
    username:
      # note! no enum here means it is an open value
      default: demo
      description: this value is assigned by the service provider, in this example `gigantic-server.com`
    port:
      enum:
        - '8443'
        - '443'
      default: '8443'
    basePath:
      # open meaning there is the opportunity to use special base paths as assigned by the provider, default is `v2`
      default: v2
```

The current version indicates a variable using angle brackets.

```yaml
servers:
  - url: 'https://<yourdomain>.api.engagement.dimelo.com/1.0'
```

### PR DESCRIPTION
**Pull request by nel**
Make variables url follow openapi spec for better website generation #19

hostname is not enum to support more than just prod.

Pity it does not work in editor.swagger.io because of CORS headers (it's intentional as I don't want to incentivize people to use global api token in the browser but it's a bit painful for this particular usecase.

Could not test bearer because of that

### CODE REVIEW

**Comment by nel** [comment]
@CedricBm I am not sure whether i should assign you to PR or simply add you as reviewer so i did both. I assign myself in another PR and got 0 mail but maybe it came from the 'myself' part.

File: specs/engage-digital_openapi3.yaml [hunk 1]
```diff
@@ -5,7 +5,15 @@ info:
   title: RingCentral Engage Digital API
   version: '1.0'
 servers:
-  - url: 'https://<yourdomain>.api.engagement.dimelo.com/1.0'
+  - url: 'https://{account_name}.{platform_hostname}/1.0'
+    description: API server endpoint
+    variables:
+      account_name:
+        default: domainname
+        description: your own account name used as subdomain, it's the same as in the url of the engage digital service.
```
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:13]: We seem to always begin our description with a capital letter :)

```suggestion
        description: Your own account name used as subdomain, it's the same as in the url of the engage digital service.
```

File: specs/engage-digital_openapi3.yaml [hunk 2]
```diff
@@ -5,7 +5,15 @@ info:
   title: RingCentral Engage Digital API
   version: '1.0'
 servers:
-  - url: 'https://<yourdomain>.api.engagement.dimelo.com/1.0'
+  - url: 'https://{account_name}.{platform_hostname}/1.0'
+    description: API server endpoint
+    variables:
+      account_name:
+        default: domainname
```
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:12]: To make it extra clear that hyphens are allowed.

```suggestion
        default: domain-name
```

File: specs/engage-digital_openapi3.yaml [hunk 3]
```diff
@@ -5,7 +5,15 @@ info:
...
-  - url: 'https://<yourdomain>.api.engagement.dimelo.com/1.0'
+  - url: 'https://{account_name}.{platform_hostname}/1.0'
+    description: API server endpoint
+    variables:
+      account_name:
+        default: domainname
+        description: your own account name used as subdomain, it's the same as in the url of the engage digital service.
+      platform_hostname:
+        default: api.engagement.dimelo.com
+        description: Depending on environment the base hostname is changing. In production it's api.digital.ringcentral.com for NA customers and api.engagement.dimelo.comfor historical customers. It's the same base hostname as in the url of engage digital service.
```
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:16]:
```suggestion
        description: Depending on environment the base hostname is changing. In production, it's api.digital.ringcentral.com for NA customers and api.engagement.dimelo.com for historical customers. It's the same base hostname as in the url of engage digital service.
```

File: specs/engage-digital_openapi3.yaml [hunk 4]
```diff
@@ -5,7 +5,15 @@ info:
   title: RingCentral Engage Digital API
   version: '1.0'
 servers:
-  - url: 'https://<yourdomain>.api.engagement.dimelo.com/1.0'
+  - url: 'https://{account_name}.{platform_hostname}/1.0'
```
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:8]: Should we rather hardcode the `api` part here?

```suggestion
  - url: 'https://{account_name}.api.{platform_hostname}/1.0'
```

**Review by CedricBm — CHANGES_REQUESTED** [review]
I received the email notification as soon as you made me reviewer 

> @nel requested your review on: #21 api: add variables to endpoint url of openapi spec, fixes #19.

And I also received the assignation email notification

> Assigned #21 to @CedricBm.

File: specs/engage-digital_openapi3.yaml [hunk 5]
```diff
@@ -5,7 +5,15 @@ info:
...
 servers:
-  - url: 'https://<yourdomain>.api.engagement.dimelo.com/1.0'
+  - url: 'https://{account_name}.api.{platform_hostname}/1.0'
+    description: API server endpoint
+    variables:
+      account_name:
+        default: domain-name
+        description: Your own account name used as subdomain, it's the same as in the url of the engage digital service.
+      platform_hostname:
+        default: api.engagement.dimelo.com
```
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:15]: If you agreed on the `.api` hardcoded in the url, then you can remove it from here. Also from the description.

File: specs/engage-digital_openapi3.yaml [hunk 6]
```diff
@@ -3,9 +3,17 @@ info:
   description: ''
   termsOfService: 'https://developer.ringcentral.com'
   title: RingCentral Engage Digital API
-  version: '1.0'
+  version: '1.01'
```
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:6]: Is this change expected? If so, shouldn't it be `1.0.1` or `1.1`?

**Comment by CedricBm** [comment]
One last comment about the version, then lgtm

**Pull request merged**